contracts/solidity/block/BlockInfo.sol

```diff
@@ -7,8 +7,8 @@ contract BlockInfo {
         return block.basefee;
     }
 
-    function getBlockHash() public view returns (bytes32) {
-        return blockhash(block.number - 1);
+    function getBlockHash(uint256 blockNumber) public view returns (bytes32) {
```

Comment:
Added `blockNumber` param for consistency because `block.number - 1` depends on the current block.